src/events/ready.ts

```diff
@@ -6,24 +6,28 @@ export default async function ReadyEvent(discordClient: DiscordClient) {
 	Logger.info(`Ready to serve, found ${discordClient.guilds.cache.size} guilds`);
 
 	discordClient.guilds.cache.forEach(guild => {
-		Logger.info(`Checking for guild ${guild.id}`);
-		ServerSettingsRepository.GetByGuildId(guild.id).then(serverSettings => {
-			if (serverSettings === null) {
-				ServerSettingsRepository.Save({
-					id: 0,
-					guildId: guild.id,
-					deleted: null,
-					prefix: ';',
-					logChannel: null,
-					modLogChannel: null,
-					systemNotice: true,
-					streamLiveRole: null,
-					streamShout: null,
-					adminRole: null, 
-					moderatorRole: null,
-				});
-				Logger.info(`Created guild ${guild.id}`);
-			}
-		});
+		CheckGuild(guild.id)
```

Comment:
CheckGuild in ReadyEvent returns a lost promise. Maybe something like this will resolve that issue

```
await Promise.all(discordClient.guilds.cache.map(guild => CheckGuild(guild.id));
```

Reply:
I see, so we use Promise.all() to wait for all async operations to be ended and map() instead of foreach() to have an 'end' to it, despite having no return value.